File: pancakeSort.java
class Solution {
    public List<Integer> pancakeSort(int[] arr) {
            List<Integer> ls=new ArrayList<Integer>();
            for(int i=arr.length;i>0;i--){
                int index=findTarget(arr,i);
                if(index==-1)continue;
                else{
                    flipArray(arr,index);
                    flipArray(arr,i-1);
                    ls.add(index+1);
                    ls.add(i);
                }
            }
return ls;     
    }
     private static  int findTarget(int[]arr,int target){
            for(int i=0;i<arr.length;i++){
                if(arr[i]==target)return i;
            }
            return -1;
    }
    private static void flipArray(int []arr,int j){
        int i=0;
        while (i<j){
        int temp=arr[i];
        arr[i++]=arr[j];
        arr[j--]=temp;
        }
    }

}

File: pancakeSort.py
class Solution:
    def swapArr(self,ind:int,arr:List[int]):
        j=0
        print('henok')
        while j<ind:
            temp=arr[j]
            arr[j]=arr[ind]
            arr[ind]=temp
            j+=1
            ind-=1
    
    def pancakeSort(self, arr: List[int]) -> List[int]:
        maxlen=len(arr)
        maxInd=0
        maxval=arr[0]
        sol=[]
        while maxlen>1:
            for i in range(maxlen):
                if arr[i]>maxval:
                    maxval=arr[i]
                    maxInd=i
            print(maxval)
            self.swapArr(maxInd,arr)
            sol.append(maxInd+1)
            self.swapArr(maxlen-1,arr)
            sol.append(maxval)
            maxlen-=1
            maxval=0
        return sol
            
